tests/scene/test_node_3d.h

```diff
@@ -0,0 +1,288 @@
+		CHECK_EQ(test_child->get_global_position(), Vector3(3, 1, 3));
+
+		Basis basis = Basis(Vector3(1, 0, 0), Vector3(0, 0, 1), Vector3(0, 1, 0));
+		test_node->set_global_transform(Transform3D(basis, Vector3(3, 0, 3)));
+		CHECK_EQ(test_node->get_global_position(), Vector3(3, 0, 3));
+		CHECK_EQ(test_child->get_global_position(), Vector3(4, 1, 4));
+	}
+
+	memdelete(test_child);
+	memdelete(test_node);
+}
+
+TEST_CASE("[SceneTree][Node3D] Utility methods") {
+	Node3D *test_node1 = memnew(Node3D);
+	Node3D *test_node2 = memnew(Node3D);
+	Node3D *test_node3 = memnew(Node3D);
+	Node3D *test_sibling = memnew(Node3D);
+	SceneTree::get_singleton()->get_root()->add_child(test_node1);
+
+	test_node1->set_position(Vector3(100, 0, 100));
+	test_node1->set_rotation(Vector3(0, 0, 0));
+	test_node1->set_scale(Vector3(1, 1, 1));
+	test_node1->add_child(test_node2);
+
+	test_node2->set_position(Vector3(100, 0, 0));
+	test_node2->set_rotation(Vector3(0, 0, 0));
+	test_node2->set_scale(Vector3(1, 1, 1));
+	test_node2->add_child(test_node3);
+	test_node2->add_child(test_sibling);
+
+	test_node3->set_position(Vector3(0, 0, 10));
+	test_node3->set_rotation(Vector3(0, 0, 0));
+	test_node3->set_scale(Vector3(2, 2, 2));
+
+	test_sibling->set_position(Vector3(5, 0, 10));
+	test_sibling->set_rotation(Vector3(0, 0, 0));
+	test_sibling->set_scale(Vector3(2, 2, 1));
+
+	SUBCASE("[Node3D] look_at") {
+		test_node3->look_at(Vector3(1, 0, 1));
+
+		CHECK(test_node3->get_global_position().is_equal_approx(Vector3(200, 0, 110)));
+		CHECK(test_node3->get_global_rotation().is_equal_approx(Vector3(0, 1.069691, 0)));
+		CHECK(test_node3->get_position().is_equal_approx(Vector3(0, 0, 10)));
+		CHECK(test_node3->get_rotation().is_equal_approx(Vector3(0, 1.069691, 0)));
+		CHECK(test_node3->get_scale().is_equal_approx(Vector3(2, 2, 2)));
+
+		test_node3->look_at(Vector3(0, 0, 10));
+
+		CHECK(test_node3->get_global_position().is_equal_approx(Vector3(200, 0, 110)));
+		CHECK(test_node3->get_global_rotation().is_equal_approx(Vector3(0, 1.107149, 0)));
+		CHECK(test_node3->get_position().is_equal_approx(Vector3(0, 0, 10)));
+		CHECK(test_node3->get_rotation().is_equal_approx(Vector3(0, 1.107149, 0)));
+		CHECK(test_node3->get_scale().is_equal_approx(Vector3(2, 2, 2)));
+
+		// Revert any rotation caused by look_at
+		test_node3->set_rotation(Vector3(0, 0, 0));
+
+		// look_at non-local target
+		Node3D *target_node = memnew(Node3D);
+		SceneTree::get_singleton()->get_root()->add_child(target_node);
+
+		target_node->set_position(Vector3(10, 10, 10));
+		test_node3->look_at(target_node->get_global_position());
+		CHECK(test_node3->get_global_rotation().is_equal_approx(Vector3(0.093183, 1.086318, 0)));
+
+		test_node3->set_rotation(Vector3(0, 0, 0));
+		memdelete(target_node);
+	}
+
+	SUBCASE("[Node3D] rotate_object_local") {
+		// Basic rotate_object_local tests
+		test_node3->rotate_object_local(Vector3(0, 1, 0), real_t(Math::deg_to_rad(180.00)));
```

Comment:
```suggestion
		test_node3->rotate_object_local(Vector3(0, 1, 0), Math_PI);
```
Since these methods use radians it'd be best to remove an error factor based on conversion IMO

Reply:
Yes, that is much smoother.